openmc/cell.py

```diff
@@ -344,6 +353,11 @@ def distribcell_paths(self, distribcell_paths):
                                string_types)
         self._distribcell_paths = distribcell_paths
 
+    @num_instances.setter
+    def num_instances(self, num_instances):
```

Comment:
I would argue that we shouldn't even have a setter for `num_instances` since we wouldn't want a user setting it themselves. Can we just have the `count` methods directly set the private attribute?